### PR TITLE
coinbase/kraken: Add IOC support to SubmitOrder wrappers

### PR DESCRIPTION
**Pull request by geseq**
# PR Description

Similar to #1022 adds IOC support to SubmitOrder wrapper for Kraken (ws and rest) and Coinbase.

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [X] New feature (non-breaking change which adds functionality)

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [x] go test ./... -race
- [x] golangci-lint run
- [x] Manually by using exchanges package directly

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes

### CODE REVIEW

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for your contribution!

File: exchanges/coinbasepro/coinbasepro_wrapper.go [hunk 1]
```diff
@@ -554,12 +554,16 @@ func (c *CoinbasePro) SubmitOrder(ctx context.Context, s *order.Submit) (*order.
...
+		if s.ImmediateOrCancel {
+			timeInForce = string(CoinbaseRequestParamsTimeIOC)
+		}
 		orderID, err = c.PlaceLimitOrder(ctx,
 			"",
 			s.Price,
 			s.Amount,
 			s.Side.Lower(),
-			"",
+			timeInForce,
```
Comment (shazbert) [inline exchanges/coinbasepro/coinbasepro_wrapper.go:566]: Change type from string to `RequestParamsTimeForceType` to be explicit or alternatively you can to reduce type conversions:

```go
const CoinbaseRequestParamsTimeIOC = "IOC"
```

File: exchanges/kraken/kraken_types.go [hunk 2]
```diff
@@ -409,6 +409,7 @@ type AddOrderOptions struct {
 	ClosePrice     float64
 	ClosePrice2    float64
 	Validate       bool
+	TimeInForce    string
```
Comment (shazbert) [inline exchanges/kraken/kraken_types.go:412]: Here as well. Either remove type or force usage of type.

**Review by shazbert — CHANGES_REQUESTED** [review]
Thanks, just a pedantic change request.

File: exchanges/kraken/kraken_types.go (hunk 2 above)
Comment (geseq) [reply exchanges/kraken/kraken_types.go]: done

File: exchanges/coinbasepro/coinbasepro_wrapper.go (hunk 1 above)
Comment (geseq) [reply exchanges/coinbasepro/coinbasepro_wrapper.go:566]: done

File: exchanges/coinbasepro/coinbasepro_wrapper.go [hunk 3]
```diff
@@ -554,12 +554,16 @@ func (c *CoinbasePro) SubmitOrder(ctx context.Context, s *order.Submit) (*order.
 			fpair.String(),
 			"")
 	case order.Limit:
+		timeInForce := CoinbaseRequestParamsTimeGTC
```
Comment (shazbert) [inline exchanges/coinbasepro/coinbasepro_wrapper.go:557]: There seems to be a json tag issue in the outbound struct for `time in force`, it's sending it as `time_in_foce` 😆 
```
CoinbasePro request body: {"price":"1","product_id":"BTC-USD","side":"buy","size":"1","time_in_foce":"IOC","type":"limit"}
```

File: exchanges/kraken/kraken_wrapper.go [hunk 4]
```diff
@@ -726,14 +726,19 @@ func (k *Kraken) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Submi
 	status := order.New
 	switch s.AssetType {
 	case asset.Spot:
+		timeInForce := KrakenRequestParamsTimeGTC
```
Comment (shazbert) [inline exchanges/kraken/kraken_wrapper.go:729]: Looking good and defaults to GTC when false. 💐 
```
Kraken attempt 1 request path: https://api.kraken.com/0/private/AddOrder
Kraken request header [Api-Key]: [sup bros]
Kraken request header [Api-Sign]: [qjOVCOtTCuoxlTyz6Fzue7KYiDIqw2C3l1xSDizcys0GDASi67rUfcfz4BhrkKMGDVGFYpsCB/YOB1Im1TL9Rw==]
Kraken request type: POST
Kraken request body: &{nonce=1663201369768872200&ordertype=limit&pair=XBTUSD&price=1&timeinforce=IOC&type=buy&volume=1 0 -1}
HTTP status: 200 OK, Code: 200
```

**Review by shazbert — CHANGES_REQUESTED** [review]
Looking good just one random issue with coinbase. I don't have any funds on those exchanges so it was just eyeballing outbound tests.

File: exchanges/coinbasepro/coinbasepro_wrapper.go (hunk 3 above)
Comment (geseq) [reply exchanges/coinbasepro/coinbasepro_wrapper.go:557]: good catch! it seems to have been incorrect in master.

**Review by shazbert — APPROVED** [review]
tested ACK. Thanks!

```
CoinbasePro attempt 1 request path: https://api.pro.coinbase.com/orders
CoinbasePro request header [Cb-Access-Timestamp]: [1663546653]
CoinbasePro request header [Cb-Access-Key]: [hello moto]
CoinbasePro request header [Cb-Access-Passphrase]: [1337]
CoinbasePro request header [Content-Type]: [application/json]
CoinbasePro request header [Cb-Access-Sign]: [GdMCCwXoq8nmCr8+5361Y1z+RWBpHdJArmQazlJWtJ4=]
CoinbasePro request type: POST
CoinbasePro request body: {"price":"1","product_id":"BTC-USD","side":"buy","size":"1","time_in_force":"IOC","type":"limit"}
```

File: exchanges/coinbasepro/coinbasepro.go [hunk 5]
```diff
@@ -334,7 +334,7 @@ func (c *CoinbasePro) GetHolds(ctx context.Context, accountID string) ([]Account
 // timeInforce - [optional] GTC, GTT, IOC, or FOK (default is GTC)
 // cancelAfter - [optional] min, hour, day * Requires time_in_force to be GTT
 // postOnly - [optional] Post only flag Invalid when time_in_force is IOC or FOK
-func (c *CoinbasePro) PlaceLimitOrder(ctx context.Context, clientRef string, price, amount float64, side, timeInforce, cancelAfter, productID, stp string, postOnly bool) (string, error) {
+func (c *CoinbasePro) PlaceLimitOrder(ctx context.Context, clientRef string, price, amount float64, side string, timeInforce RequestParamsTimeForceType, cancelAfter, productID, stp string, postOnly bool) (string, error) {
```
Comment (gloriousCode) [inline exchanges/coinbasepro/coinbasepro.go:337]:
```suggestion
func (c *CoinbasePro) PlaceLimitOrder(ctx context.Context, clientRef string, price, amount float64, side, cancelAfter, productID, stp string, timeInforce RequestParamsTimeForceType, postOnly bool) (string, error) {
```

**Review by gloriousCode — CHANGES_REQUESTED** [review]
The most tiny change request :smile:

**Review by gloriousCode — APPROVED** [review]
I rescind the change request, because `cancelAfter` string becomes weird in parameter order...

tACK!

**Review by thrasher- — APPROVED** [review]
utACK (no balance on exchanges), thanks for the PR! Nice catch also @shazbert 👀

**Pull request merged**